doc/manual/documentation.rst

```diff
@@ -0,0 +1,115 @@
+.. _man-documentation:
+
+***************
+ Documentation
+***************
+
+Julia enables package developers and users to document functions, types and
+other objects easily, either via the built-in documentation system in Julia 0.4
+or the `Docile.jl <http://github.com/MichaelHatherly/Docile.jl>`_ package in
+Julia 0.3.
+
+.. doctest::
+
+    VERSION < v"0.4-" && using Docile
+
+    @doc doc"Tells you if there are too foo items in the array." ->
+    foo(xs::Array) = ...
+
+Documentation is interpreted as `Markdown <http://en.wikipedia.org/wiki/Markdown>`_,
+so you can use indentation and code fences to delimit code examples from text.
```

Comment:
On the latter point (automatically including the syntax), I see you've argued elsewhere that perhaps it's better not to do this. I can see your point.